src/modules/esl-traversing-query/core/esl-traversing-query.ts

```diff
@@ -96,7 +96,34 @@ export class TraversingQuery {
     return uniq(result);
   }
 
-  static traverse(query: string, findFirst: boolean, base?: Element | null, scope: Element | Document = document): Element[] {
+  /** Split multiple queries separated by comma (respects query brackets) */
+  public static splitQueries(str: string): string[] {
+    let last = 0;
+    let stack = 0;
+    const result = [];
+    for (let i = 0; i < str.length; i++) {
+      if (str[i] === '(') stack++;
+      if (str[i] === ')') stack = Math.max(0, stack - 1);
+      if (str[i] === ',' && !stack) {
+        result.push(str.substring(last, i).trim());
+        last = i + 1;
+      }
+    }
```

Comment:
```suggestion
    [...str].forEach((char: string, index: number) => {
      if (char === '(') stack++;
      if (char === ')') stack = Math.max(0, stack - 1);
      if (char === ',' && !stack) {
        result.push(str.substring(last, index).trim());
        last = index + 1;
      }
    });
```

Reply:
It does not speed up code execution.
As for production projects, that's a cool solution.
As for ESL that's not so cool as it requires more operations than in the original implementation.
Also, there is a problem that it requires ES6 iterators, which is part of the Symbol standard, but we still should support ES5 for 4-th version of ESL.